utils/common/src/main/java/org/apache/brooklyn/test/Asserts.java

```diff
@@ -1381,4 +1381,18 @@ public static void assertSize(Iterable<?> list, int expectedSize) {
         if (Iterables.size(list)!=expectedSize) fail("List has wrong size "+Iterables.size(list)+" (expected "+expectedSize+"): "+list);
     }
 
+    public static void assertInstanceOf(Object obj, Class<?> type) {
+        assertThat(obj, Predicates.instanceOf(type));
```

Comment:
Not convinced that this is useful (thought agree it can be made so, to make the test more explicit). It gives less information about the problem than a good old fasioned `ClassCastException`: it doesn't tell you what type `obj` is (just its `toString`).